packages/graphql/src/type-maps.ts

```diff
@@ -101,3 +159,143 @@ export abstract class TypeMap<T extends Type, G extends GraphQLType> {
    */
   protected abstract materialize(context: TSPContext<T>): G;
 }
+
+/**
+ * TypeMap for GraphQL Object types (output types)
```

Comment:
Is there some shared functionality here between output object and input object types that warrants a common parent class?